### PR TITLE
Enable publishing in VMR

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Contributes to https://github.com/dotnet/source-build/issues/4101

Enables publishing in VMR.

Installers are not collected by common arcade publishing infra. All repos that build installers need to collect them in Publishing.props file.

Additionally installers need to be published to correct blob path, which includes runtime package version. I'm obtaining this from dotnet-runtime-<Version-<Rid>.<Extension> archive.

### CODE REVIEW

File: eng/Publishing.props [hunk 1]
```diff
@@ -2,6 +2,60 @@
...
+      <_RuntimeVersion>$(_RuntimeArchiveFilename.Replace('$(_RuntimeFilenamePrefix)','').Replace('-$(_PackageRid)$(ArchiveExtension)',''))</_RuntimeVersion>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <!-- Do not push .nupkg files from Linux and macOS builds. They'll be packed up separately and signed on Windows.
+           Do not remove if post build sign is true, as we avoid the xplat codesign jobs, and need to have
+           the nupkgs pushed. Do not do this if building from source, since we want the source build intermediate package
+           to be published. Use DotNetBuildRepo as DotNetBuildFromSource is only set in the internal source build,
+           and Build.proj is invoked from the wrapper build. -->
+      <ItemsToPushToBlobFeed Remove="@(ItemsToPushToBlobFeed)" Condition="'$(OS)' != 'Windows_NT' and '$(PostBuildSign)' != 'true' and '$(DotNetBuildRepo)' != 'true'" />
```
Comment (mmitche) [inline eng/Publishing.props:51]: This should not conditionalized based on OS.

File: eng/Publishing.props [hunk 2]
```diff
@@ -2,6 +2,60 @@
 
   <PropertyGroup>
     <ProducesDotNetReleaseShippingAssets>true</ProducesDotNetReleaseShippingAssets>
+    <PublishDependsOnTargets>$(PublishDependsOnTargets);_PublishInstallers</PublishDependsOnTargets>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (mmitche) [inline eng/Publishing.props:8]: Please hoist the condition into the <ItemGroup>

File: eng/Publishing.props [hunk 3]
```diff
@@ -2,6 +2,60 @@
...
+    <_InstallersToPublish Remove="@(_InstallersToPublish)" />
+    <_InstallersToPublish Include="$(ArtifactsPackagesDir)**\*.tar.gz" UploadPathSegment="Runtime" Condition="'$(DotNetBuildRepo)' == 'true'" />
+    <_InstallersToPublish Include="$(ArtifactsPackagesDir)**\*.zip" UploadPathSegment="Runtime" Condition="'$(DotNetBuildRepo)' == 'true'" />
+    <_InstallersToPublish Include="$(ArtifactsPackagesDir)**\*.deb" UploadPathSegment="Runtime" Condition="'$(DotNetBuildRepo)' == 'true'" />
+    <_InstallersToPublish Include="$(ArtifactsPackagesDir)**\*.rpm" UploadPathSegment="Runtime" Condition="'$(DotNetBuildRepo)' == 'true'" />
+    <_InstallersToPublish Include="$(ArtifactsPackagesDir)**\*.exe" UploadPathSegment="Runtime" Condition="'$(DotNetBuildRepo)' == 'true'" />
+    <_InstallersToPublish Include="$(ArtifactsPackagesDir)**\*.msi" UploadPathSegment="Runtime" Condition="'$(DotNetBuildRepo)' == 'true'" />
+  </ItemGroup>
+
+  <Target Name="_PublishInstallers">
```
Comment (mmitche) [inline eng/Publishing.props:18]: I'm still not sure you need the target in runtime's case, in the VMR build. What happens in the VMR build when -publish runs from runtime?

Comment (NikolaMilosavljevic) [reply eng/Publishing.props:18]: Everything gets published properly from `runtime` repo, in VMR build. Target is just a convenience I think, and it is modeled on `aspnetcore` implementation. I did not see any issues in binlogs.

File: eng/Publishing.props (hunk 2 above)
Comment (NikolaMilosavljevic) [reply eng/Publishing.props:8]: I presume the ask is to move the condition from individual items to the group - `Condition="'$(DotNetBuildRepo)' == 'true'"`. I'll do that.

File: eng/Publishing.props (hunk 1 above)
Comment (mmitche) [reply eng/Publishing.props:51]: We should not.

Comment (NikolaMilosavljevic) [reply eng/Publishing.props:51]: OK - will remove this and update the PR.

Comment (NikolaMilosavljevic) [reply eng/Publishing.props]: Fixed with https://github.com/dotnet/runtime/pull/99432/commits/18768165afb5bd3020cf0e6cc56c104265cdc7ba

File: eng/Publishing.props (hunk 2 above)
Comment (NikolaMilosavljevic) [reply eng/Publishing.props]: Fixed with https://github.com/dotnet/runtime/pull/99432/commits/18768165afb5bd3020cf0e6cc56c104265cdc7ba

**Review by ViktorHofer — APPROVED** [review]

**Review by jkoritzinsky — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Verified changes offline 👍

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**